config/zephyr/chip-module/Kconfig.defaults

```diff
@@ -40,9 +40,11 @@ config LOG_DEFAULT_LEVEL
 
 endif
 
+# disable synchronous printk to avoid blocking IRQs which
+# may affect time sensitive components
 config PRINTK_SYNC
 	bool
-	default y
+	default n
```

Comment:
out of curiosity, in which scenario did you face issue with this ?

Reply:
Fortunately, it didn't impact the WiFi driver (but it may occur in some specific scenarios). We faced a very sporadic issue with 802.15.4 radio when that kconfig was enabled.